### PR TITLE
Compact space workflow prompts

### PR DESCRIPTION
**Pull request by lsm**
Refactors shared reviewer/QA prompt contracts, trims Space coordinator and workflow prompts, and caps dynamic task activation context to reduce prompt bloat while preserving workflow semantics.

Validation:
- bun run check
- bun test tests/unit/5-space/agent/custom-agent.test.ts tests/unit/5-space/other/seed-agents.test.ts tests/unit/5-space/runtime/space-chat-agent.test.ts tests/unit/1-core/agent/query-options-builder.test.ts tests/unit/5-space/workflow/built-in-workflows.test.ts
- bun test tests/online/space/space-happy-path-code-review.test.ts (passes; SDK subprocess logs expected model/account mismatch in mocked happy-path tail)

### CODE REVIEW

File: packages/daemon/src/lib/space/agents/custom-agent.ts [hunk 1]
```diff
@@ -348,35 +355,33 @@ export function buildCustomAgentTaskMessage(config: CustomAgentConfig): string {
...
   const standingLines: string[] = [];
   if (space.instructions?.trim()) standingLines.push(space.instructions.trim());
   if (workflow?.instructions?.trim()) standingLines.push(workflow.instructions.trim());
   if (standingLines.length > 0) {
     sections.push('');
     sections.push('## Standing Instructions');
     sections.push('');
-    sections.push(standingLines.join('\n\n'));
+    sections.push(
+      truncateContextBlock(standingLines.join('\n\n'), STANDING_INSTRUCTIONS_PROMPT_LIMIT)
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/agents/custom-agent.ts]: Fixed in 077b0ddf3. Standing Instructions now render the full combined space + workflow policy without truncation, so mandatory operator/workflow rules remain visible to every task agent.

File: packages/daemon/src/lib/space/agents/custom-agent.ts [hunk 2]
```diff
@@ -419,6 +424,62 @@ function truncateMemoryPromptContent(content: string): string {
...
+): string[] {
+  if (!items || items.length === 0) return [];
+  const lines: string[] = [];
+  let used = 0;
+  let omitted = 0;
+  for (const item of [...items].reverse()) {
+    const line = `- ${item}`;
+    if (used + line.length + 1 > totalLimit) {
+      omitted += 1;
+      continue;
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/agents/custom-agent.ts]: Fixed in 077b0ddf3. The previous-work builder now always keeps the newest summary visible; if the newest summary alone exceeds the budget, it truncates that item instead of dropping it. Added regression coverage for the oversized-newest-summary case.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** Claude Sonnet 4 | **Client:** NeoKai | **Provider:** Anthropic

**Verdict: APPROVE**

All 9 audit recommendations implemented correctly. Net -400 lines. 561 focused tests pass. `make check` clean. No semantic changes.

**Rec coverage verified:**
1. ✅ Reviewer contract centralized in `system-contracts.ts`
2. ✅ Terminal-action semantics enriched in MCP tool descriptions/schemas
3. ✅ GitHub review procedure compacted into system contract
4. ✅ Space coordinator prompt trimmed (-239 lines)
5. ✅ Dynamic context sections capped with newest-first priority and truncation fallback
6. ✅ FULLSTACK_REVIEW_PROMPT references Reviewer System Contract
7. ✅ Worktree isolation prompt trimmed (-19 lines)
8. ✅ PD_PLAN_REVIEW_PROMPT references Reviewer System Contract
9. ✅ FULLSTACK_QA_PROMPT uses QA_SYSTEM_CONTRACT

**Fix commit (077b0ddf3) verified:**
- Standing Instructions no longer truncated (space + workflow policy always rendered in full)
- Oversized newest previous-work summary truncated rather than dropped
- `truncateBulletLine` edge-case helper added
- All 561 tests pass
---REVIEW_POSTED---
url: <pending>
recommendation: APPROVE
p0: 0
p1: 0
p2: 0
p3: 0
summary: Clean implementation of all 9 prompt audit recommendations with prior review feedback addressed. 561 tests pass, make check clean, net -400 lines, zero semantic changes.
---END_REVIEW_POSTED---

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** Claude Sonnet 4 | **Client:** NeoKai | **Provider:** Anthropic

**Verdict: REQUEST_CHANGES**

Three items need addressing:

### 1. Remove dynamic context capping (P1)
The `buildCappedBulletLines`, `truncateContextBlock`, `buildRelevantMemoryLines` functions and their four budget constants (`PREVIOUS_WORK_TOTAL_LIMIT=900`, `RELEVANT_MEMORY_PROMPT_TOTAL_LIMIT=500`, `PROJECT_CONTEXT_PROMPT_LIMIT=1200`, `STANDING_INSTRUCTIONS_PROMPT_LIMIT=1200`) should be reverted. Go back to unbounded previous work, relevant memories, project context. The oversized-newest-summary truncation edge case can stay.

### 2. Restore `gh api` example in reviewer system contract (P1)
The explicit bash examples for posting reviews via REST API and inline comments need to come back to `REVIEWER_SYSTEM_CONTRACT` in `system-contracts.ts`. Without concrete syntax, reviewer agents may fumble the invocation.

### 3. Restore detailed review process + emphasize multiple subagents (P1)
Bring back the numbered procedural review steps (read task/PR, dispatch sub-agents, read files completely, validate APIs, evaluate holistically). Key emphasis: the reviewer should dispatch **multiple** Task general-purpose sub-agents to cover different aspects (callers/callees, tests, integration risks, security, etc.), forming review aspects based on the specific task. Each round should be a fresh from-scratch review, not incremental.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** Claude Sonnet 4 | **Client:** NeoKai | **Provider:** Anthropic

### Three requested changes — all verified ✅

1. **Context capping removed** — `buildCappedBulletLines`, `truncateContextBlock`, `buildRelevantMemoryLines` and their budget constants reverted. Previous work, relevant memories, project context all unbounded again. Oversized-newest-summary truncation guard preserved. Tests pass.

2. **`gh api` examples restored** — Full bash examples for top-level review, own-PR fallback, and anchored inline comments present in `REVIEWER_SYSTEM_CONTRACT` (lines 29-67). Concrete syntax available for reviewer agents.

3. **Review process + multi-subagent emphasis restored** — Six numbered process steps (lines 9-15), explicit "Dispatch multiple Task general-purpose sub-agents" with aspect examples (callers/callees, tests, integration, security, API contracts, data migrations, performance, UX), "each review round is fresh from scratch" (line 5). All present.

### Scope creep — P1 concern

Commit `0641a41c3` adds **+2,923 lines across 35 files**, far exceeding the three requested prompt changes. Unrelated additions mixed into this PR:

| Unrelated change | Files |
|---|---|
| Validation-only completion node + gate | `built-in-workflows.ts`, tests |
| `isWorkflowTerminalNode` detection | `task-agent-manager.ts`, tests |
| Post-approval pending state cleanup | `post-approval-router.ts`, `space-task-manager.ts`, `end-node-handlers.ts` |
| Evolution scope cross-posting evidence | `evolution-scope-service.ts`, `evolution-repository.ts`, tests |
| Gate script env key allowlist | `gate-script-executor.ts`, tests |
| Sibling quiesce by source node | `space-runtime.ts` |
| Flaky test runner infrastructure | `scripts/` (4 new files) |

These are separate features/fixes. Mixing them in one PR:
- Makes atomic revert impossible (validation-only node can't be reverted without reverting prompt fixes)
- Inflates review surface from ~60 lines (the three prompt changes) to ~3,000 lines
- Hides bugs behind volume

**Recommendation:** Separate into 2-3 PRs. This PR keeps only the prompt contract changes (system-contracts, custom-agent, seed-agents, built-in-workflows prompt text, query-options-builder, space-chat-agent, tool descriptions). The validation-only node, terminal node detection, post-approval state, evolution cross-posting, and flaky test runner each get their own PR.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** Claude Sonnet 4 | **Client:** NeoKai | **Provider:** Anthropic

**Verdict: APPROVE**

Three requested changes verified in current PR diff. Scope creep was worktree-only — PR contains 9 source files, 513 diff lines, 3 commits. 570 tests pass. `make check` clean.

1. ✅ Dynamic context capping removed — `buildCappedBulletLines`, `truncateContextBlock`, `buildRelevantMemoryLines` and budget constants gone. Previous work, memories, project context all unbounded. Oversized-newest-summary truncation guard preserved.
2. ✅ `gh api` examples restored — three concrete bash examples (top-level review, own-PR fallback, anchored inline comments) in `REVIEWER_SYSTEM_CONTRACT`.
3. ✅ Review process + multi-subagent emphasis restored — six numbered steps, "Dispatch multiple Task general-purpose sub-agents" with aspect examples, "each review round is fresh from scratch".
---REVIEW_POSTED---
url: <pending>
recommendation: APPROVE
p0: 0
p1: 0
p2: 0
p3: 0
summary: All three requested prompt changes correctly implemented. PR clean (9 files, 3 commits, 570 tests pass). Ready to merge.
---END_REVIEW_POSTED---

File: packages/daemon/src/lib/space/agents/system-contracts.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+If reviewing your own PR, GitHub rejects APPROVE/REQUEST_CHANGES. Fallback:
+
+\`\`\`bash
+gh api repos/{owner}/{repo}/pulls/{n}/reviews \
+  -f event='COMMENT' \
+  -f body='## 🤖 Review by <your model> (<your provider>)
+
+> **Model:** <your model> | **Client:** NeoKai | **Provider:** <your provider>
+
+Recommendation: REQUEST_CHANGES
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/agents/system-contracts.ts]: Fixed in 0979f0ef4. The own-PR COMMENT fallback now says `Recommendation: <APPROVE or REQUEST_CHANGES — match your actual verdict>` instead of hard-coding REQUEST_CHANGES, so approved own-PR reviews preserve the approval recommendation. Added regression coverage in seed-agents tests.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** Claude Sonnet 4 | **Client:** NeoKai | **Provider:** Anthropic

**Verdict: APPROVE**

Own-PR COMMENT fallback now uses `Recommendation: <APPROVE or REQUEST_CHANGES — match your actual verdict>` instead of hard-coded REQUEST_CHANGES. Correct fix. Regression test added. 91 tests pass.
---REVIEW_POSTED---
url: <pending>
recommendation: APPROVE
p0: 0
p1: 0
p2: 0
p3: 0
summary: Own-PR fallback verdict fix verified. 3-line change, test added. Ready to merge.
---END_REVIEW_POSTED---

File: packages/daemon/src/lib/space/agents/system-contracts.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+> **Model:** <your model> | **Client:** NeoKai | **Provider:** <your provider>
+\`\`\`
+
+GitHub review procedure: post a visible review before gate writes or terminal actions. Use REST API when you need the returned URL, with own-PR fallback from APPROVE/REQUEST_CHANGES to COMMENT while keeping the recommendation explicit in body. For line findings, post anchored PR comments and capture html_url values.
+
+Post top-level review and capture URL:
+
+\`\`\`bash
+gh api repos/{owner}/{repo}/pulls/{n}/reviews \
+  -f event='REQUEST_CHANGES' \
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/agents/system-contracts.ts]: Fixed in latest commit. The top-level review example now uses `-f event='<APPROVE|REQUEST_CHANGES>'` with a comment to match the actual verdict.

File: packages/daemon/src/lib/space/workflows/built-in-workflows.ts [hunk 5]
```diff
@@ -433,71 +417,26 @@ const PD_PLANNING_PROMPT =
...
-  'change, so the Planner can revise and re-open. Do NOT call `approve_task` or ' +
-  '`submit_for_approval` — they are reserved for the Task Dispatcher.';
+  'You are one of four independent Plan Reviewers. Review the plan PR through your lens before ' +
+  'tasks are dispatched. Use the Reviewer System Contract for review quality and severity.\n\n' +
+  'Plan Review is not the end node: do not call approve_task or submit_for_approval. Your terminal ' +
+  'action is your `approvals` vote on `plan-approval-gate`. Vote approved only for zero P0-P3 ' +
+  'lens findings; otherwise vote rejected and send actionable feedback to Planning.\n\n' +
+  'Procedure: read `gh pr diff`/`gh pr view`, post a visible PR review comment, then ' +
+  'send_message(target="Task Dispatcher", data: { approvals: { "<your lens>": "approved" }, ' +
+  'pr_url: "<plan PR url>" }). First three approvals normally get a gate-blocked response; ' +
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/workflows/built-in-workflows.ts]: Fixed in latest commit. The Plan Review vote example now includes `message: "<short summary>"` alongside the approvals data.

File: packages/daemon/src/lib/space/agents/system-contracts.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+gh api repos/{owner}/{repo}/pulls/{n}/comments \
+  -f body='<finding body>' \
+  -f commit_id='<head sha>' \
+  -f path='path/to/file.ts' \
+  -F line=123 \
+  -f side='RIGHT' \
+  --jq '.html_url'
+\`\`\`
+
+Terminal-action contract: follow approve_task/submit_for_approval tool descriptions. They are final close actions and valid only after an APPROVE verdict with zero P0-P3 findings and prior findings addressed. If findings remain, post review, send actionable upstream feedback, save result artifact, then stop.
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/space/agents/system-contracts.ts]: Fixed in latest commit. The terminal-action contract now explicitly states: 'If submit_for_approval fails (autonomy gate or error), stop — do not retry or loop the terminal action.'

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Sonnet 4 (Anthropic)

> **Model:** Claude Sonnet 4 | **Client:** NeoKai | **Provider:** Anthropic

**Verdict: APPROVE**

Latest commit (0979f0ef4) verified. PR contains 15 files — all prompt/tool-description/test changes. 570 tests pass. `make check` clean. No scope creep in the actual PR (unrelated changes are worktree-local only).

Summary of all changes across the PR:
- Extracted reviewer/QA contracts to `system-contracts.ts` with full `gh api` examples and multi-subagent review process
- Trimmed Space coordinator prompt, worktree isolation prompt, workflow slot prompts
- Removed dynamic context capping per feedback; preserved oversized-newest-summary guard
- Enriched terminal-action tool descriptions in MCP schemas
- Own-PR COMMENT fallback uses dynamic verdict
---REVIEW_POSTED---
url: <pending>
recommendation: APPROVE
p0: 0
p1: 0
p2: 0
p3: 0
summary: Final approval. 15 files, 570 tests pass. All review rounds addressed. Ready for squash merge into dev.
---END_REVIEW_POSTED---
